From a60eb9c80a274dcbee1dc7fcced3f887c5d162b6 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Thu, 16 Mar 2023 19:28:10 +0530
Subject: [PATCH 115/500] OpenWrt: 251-kconfig.patch

kconfig: owrt specifc dependencies

Signed-off-by: John Crispin <john@phrozen.org>

Change-Id: Ia363f455c1cc9ec5bd302078918eb860cc07c736
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 03ec57d20012d7466e5a6aaf33ca3f55f08e5c31)
---
 crypto/Kconfig        | 10 +++++-----
 drivers/bcma/Kconfig  |  1 +
 drivers/ssb/Kconfig   |  3 ++-
 lib/Kconfig           |  8 ++++----
 net/netfilter/Kconfig |  2 +-
 net/wireless/Kconfig  | 14 +++++++-------
 sound/core/Kconfig    |  4 ++--
 7 files changed, 22 insertions(+), 20 deletions(-)

diff --git a/crypto/Kconfig b/crypto/Kconfig
index 650b1b3620d8..b9c68e4daf7b 100644
--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -55,7 +55,7 @@ config CRYPTO_FIPS_VERSION
 	  By default the KERNELRELEASE value is used.
 
 config CRYPTO_ALGAPI
-	tristate
+	tristate "ALGAPI"
 	select CRYPTO_ALGAPI2
 	help
 	  This option provides the API for cryptographic algorithms.
@@ -64,7 +64,7 @@ config CRYPTO_ALGAPI2
 	tristate
 
 config CRYPTO_AEAD
-	tristate
+	tristate "AEAD"
 	select CRYPTO_AEAD2
 	select CRYPTO_ALGAPI
 
@@ -82,7 +82,7 @@ config CRYPTO_SIG2
 	select CRYPTO_ALGAPI2
 
 config CRYPTO_SKCIPHER
-	tristate
+	tristate "SKCIPHER"
 	select CRYPTO_SKCIPHER2
 	select CRYPTO_ALGAPI
 
@@ -91,7 +91,7 @@ config CRYPTO_SKCIPHER2
 	select CRYPTO_ALGAPI2
 
 config CRYPTO_HASH
-	tristate
+	tristate "HASH"
 	select CRYPTO_HASH2
 	select CRYPTO_ALGAPI
 
@@ -100,7 +100,7 @@ config CRYPTO_HASH2
 	select CRYPTO_ALGAPI2
 
 config CRYPTO_RNG
-	tristate
+	tristate "RNG"
 	select CRYPTO_RNG2
 	select CRYPTO_ALGAPI
 
diff --git a/drivers/bcma/Kconfig b/drivers/bcma/Kconfig
index b9558ff20830..2e9732e5c565 100644
--- a/drivers/bcma/Kconfig
+++ b/drivers/bcma/Kconfig
@@ -16,6 +16,7 @@ if BCMA
 # Support for Block-I/O. SELECT this from the driver that needs it.
 config BCMA_BLOCKIO
 	bool
+	default y
 
 config BCMA_HOST_PCI_POSSIBLE
 	bool
diff --git a/drivers/ssb/Kconfig b/drivers/ssb/Kconfig
index 34fa19d4b3f1..7d0d7048ccb7 100644
--- a/drivers/ssb/Kconfig
+++ b/drivers/ssb/Kconfig
@@ -29,6 +29,7 @@ config SSB_SPROM
 config SSB_BLOCKIO
 	bool
 	depends on SSB
+	default y
 
 config SSB_PCIHOST_POSSIBLE
 	bool
@@ -49,7 +50,7 @@ config SSB_PCIHOST
 config SSB_B43_PCI_BRIDGE
 	bool
 	depends on SSB_PCIHOST
-	default n
+	default y
 
 config SSB_PCMCIAHOST_POSSIBLE
 	bool
diff --git a/lib/Kconfig b/lib/Kconfig
index e2308ebe2165..e7c5f4bbabdf 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -460,16 +460,16 @@ config BCH_CONST_T
 # Textsearch support is select'ed if needed
 #
 config TEXTSEARCH
-	bool
+	bool "Textsearch support"
 
 config TEXTSEARCH_KMP
-	tristate
+	tristate "TEXTSEARCH KMP"
 
 config TEXTSEARCH_BM
-	tristate
+	tristate "TEXTSEARCH BM"
 
 config TEXTSEARCH_FSM
-	tristate
+	tristate "TEXTSEARCH FSM"
 
 config BTREE
 	bool
diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index cea415cc9f8f..8a0bbefb641a 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -22,7 +22,7 @@ config NETFILTER_SKIP_EGRESS
 	def_bool NETFILTER_EGRESS && (NET_CLS_ACT || IFB)
 
 config NETFILTER_NETLINK
-	tristate
+	tristate "Netfilter NFNETLINK interface"
 
 config NETFILTER_FAMILY_BRIDGE
 	bool
diff --git a/net/wireless/Kconfig b/net/wireless/Kconfig
index f620acd2a0f5..d26acb9a86dc 100644
--- a/net/wireless/Kconfig
+++ b/net/wireless/Kconfig
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config WIRELESS_EXT
-	bool
+	bool "Wireless extensions"
 
 config WEXT_CORE
 	def_bool y
@@ -12,10 +12,10 @@ config WEXT_PROC
 	depends on WEXT_CORE
 
 config WEXT_SPY
-	bool
+	bool "WEXT_SPY"
 
 config WEXT_PRIV
-	bool
+	bool "WEXT_PRIV"
 
 config CFG80211
 	tristate "cfg80211 - wireless configuration API"
@@ -204,7 +204,7 @@ config CFG80211_WEXT_EXPORT
 endif # CFG80211
 
 config LIB80211
-	tristate
+	tristate "LIB80211"
 	default n
 	help
 	  This options enables a library of common routines used
@@ -213,17 +213,17 @@ config LIB80211
 	  Drivers should select this themselves if needed.
 
 config LIB80211_CRYPT_WEP
-	tristate
+	tristate "LIB80211_CRYPT_WEP"
 	select CRYPTO_LIB_ARC4
 
 config LIB80211_CRYPT_CCMP
-	tristate
+	tristate "LIB80211_CRYPT_CCMP"
 	select CRYPTO
 	select CRYPTO_AES
 	select CRYPTO_CCM
 
 config LIB80211_CRYPT_TKIP
-	tristate
+	tristate "LIB80211_CRYPT_TKIP"
 	select CRYPTO_LIB_ARC4
 
 config LIB80211_DEBUG
diff --git a/sound/core/Kconfig b/sound/core/Kconfig
index e41818e59a15..73790c5e6852 100644
--- a/sound/core/Kconfig
+++ b/sound/core/Kconfig
@@ -17,7 +17,7 @@ config SND_DMAENGINE_PCM
 	tristate
 
 config SND_HWDEP
-	tristate
+	tristate "Sound hardware support"
 
 config SND_SEQ_DEVICE
 	tristate
@@ -40,7 +40,7 @@ config SND_UMP_LEGACY_RAWMIDI
 	  The device contains 16 substreams corresponding to UMP groups.
 
 config SND_COMPRESS_OFFLOAD
-	tristate
+	tristate "Compression offloading support"
 
 config SND_JACK
 	bool
-- 
2.34.1

